src/main/scala/org/ergoplatform/mining/CandidateGenerator.scala

```diff
@@ -710,7 +711,8 @@ object CandidateGenerator extends ScorexLogging {
             stateWithTxs.validateWithCost(
               tx,
               Some(upcomingContext),
-              maxTransactionComplexity
+              maxTransactionComplexity,
+              maxTransactionCost
             ) match {
```

Comment:
I think that letting a transaction with cost over limit into a candidate is also wrong

Reply:
The limit is for mempool, to protect from it spam, and prevent broadcasting too heavy transactions. For blocks, still there's another limit (cost per block) only, and miner can include transactions which exceed maxTransactionCost directly.